ui/shared/ad/CustomAdBanner.tsx

```diff
@@ -0,0 +1,61 @@
+import { Flex, chakra, Tooltip, Image } from '@chakra-ui/react';
+import { useQuery } from '@tanstack/react-query';
+import React, { useState, useEffect } from 'react';
+
+import type { AdCustomConfig } from 'types/client/ad';
+
+import config from 'configs/app';
+import type { ResourceError } from 'lib/api/resources';
+import useFetch from 'lib/hooks/useFetch';
+
+import useIsMobile from '../../../lib/hooks/useIsMobile';
+
+const CustomAdBanner = ({ className }: { className?: string }) => {
+  const isMobile = useIsMobile();
+
+  const feature = config.features.adsBanner;
+  const configUrl = (feature.isEnabled && feature.provider === 'custom') ? feature.configUrl : '';
+
+  const apiFetch = useFetch();
+  const { data: adConfig } = useQuery<unknown, ResourceError<unknown>, AdCustomConfig>(
+    [ 'custom-configs' ],
+    async() => apiFetch(configUrl),
+    {
+      enabled: feature.isEnabled && feature.provider === 'custom',
+      staleTime: Infinity,
+    });
+  const interval = adConfig?.interval || 60000;
+  const banners = adConfig?.banners || [];
+
+  const [ currentBannerIndex, setCurrentBannerIndex ] = useState(0);
```

Comment:
maybe it would be better in general to start with a random one and then change banners randomly to provide more homogenous distribution?

Reply:
That's a good suggestion, but considering that users may want a fixed order, I think it's possible to start randomly and provide a parameter to control whether the next one will be played sequentially or randomly